examples/sampling/pyg/node_classification.py

```diff
@@ -0,0 +1,268 @@
+    # Create a datapipe for mini-batch sampling with a specific neighbor fanout.
+    # Here, [10, 10, 10] specifies the number of neighbors sampled for each node at each layer.
+    # We're using `sample_neighbor` for consistency with DGL's sampling API.
+    # Note: GraphBolt offers additional sampling methods, such as `sample_layer_neighbor`,
+    # which could provide further optimization and efficiency for GNN training.
+    # Users are encouraged to explore these advanced features for potentially improved performance.
+
+    # Initialize an ItemSampler to sample mini-batches from the dataset.
+    datapipe = gb.ItemSampler(
+        dataset_set, batch_size=1024, shuffle=is_train, drop_last=is_train
+    )
+    # Sample neighbors for each node in the mini-batch.
+    datapipe = datapipe.sample_neighbor(graph, [10, 10, 10])
+    # Fetch node features for the sampled subgraph.
+    datapipe = datapipe.fetch_feature(feature, node_feature_keys=["feat"])
+    # Copy the data to the specified device.
+    datapipe = datapipe.copy_to(device=device)
+    # Create and return a DataLoader to handle data loading.
+    dataloader = gb.DataLoader(datapipe, num_workers=0)
+
+    return dataloader
+
+
+def compute_accuracy(logits, labels):
+    _, predicted = torch.max(logits, dim=1)
+    correct = (predicted == labels).sum().item()
+    return correct / labels.size(0)
+
+
+def train(model, dataloader, optimizer, criterion, device):
+    #####################################################################
+    # (HIGHLIGHT) Train the model for one epoch.
+    #
+    # - Iterates over the data loader, fetching mini-batches of graph data.
+    # - For each mini-batch, it performs a forward pass, computes loss, and
+    #   updates the model parameters.
+    # - The function returns the average loss and accuracy for the epoch.
+    #
+    # Parameters:
+    #   model: The GraphSAGE model.
+    #   dataloader: DataLoader that provides mini-batches of graph data.
+    #   optimizer: Optimizer used for updating model parameters.
+    #   criterion: Loss function used for training.
+    #   device: The device (CPU/GPU) to run the training on.
+    #####################################################################
+
+    model.train()  # Set the model to training mode
+    total_loss = 0  # Accumulator for the total loss
+    total_correct = 0  # Accumulator for the total number of correct predictions
+    total_samples = 0  # Accumulator for the total number of samples processed
+    num_batches = 0  # Counter for the number of mini-batches processed
+
+    for minibatch in dataloader:
+        node_features = minibatch.node_features["feat"]
+        labels = minibatch.labels
+        optimizer.zero_grad()
+        out = model(minibatch.blocks, node_features, device)
+        loss = criterion(out, labels)
+        total_loss += loss.item()
+        total_correct += compute_accuracy(out, labels) * labels.size(0)
```

Comment:
Is it possible to use `MF.accuracy` here?

Reply:
for sure